compat/mingw.c

```diff
@@ -383,6 +383,21 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+static int do_unlink(const char *pathname, const wchar_t *wpathname,
+		     int (*unlinker)(const wchar_t *))
+{
+	char temppath[MAX_PATH];
+	wchar_t wtemppath[MAX_LONG_PATH];
+	strncpy(temppath, pathname, sizeof(temppath) - 8);
+	strcat(temppath, ".XXXXXX");
+	mktemp(temppath);
+	if (xutftowcs_long_path(wtemppath, temppath) >= 0) {
+		if (MoveFileExW(wpathname, wtemppath, MOVEFILE_REPLACE_EXISTING | MOVEFILE_COPY_ALLOWED))
```

Comment:
After reviewing the documentation of `MOVEFILE_REPLACE_EXISTING` and `MOVEFILE_COPY_ALLOWED`, I think we should not use either. We should take special care that we do not overwrite existing files. After all, we are writing to the work tree, the contents of which are not under our control.

Reply:
You're right. I removed the flags. Now that we name it .unlink_XXXXX, it is *extremely unlikely* that a file with this name exists, and even if it does, we only fail if the original file is occupied, so I doubt we should try harder...